file.go

```diff
@@ -19,19 +19,28 @@ var (
 )
 
 // TOMLLoader satisifies the loader interface. It loads the configuration from
-// the given toml file.
+// the given toml file or Reader
 type TOMLLoader struct {
-	Path string
+	Path   string
+	Reader io.Reader
 }
 
 // Load loads the source into the config defined by struct s
+// Defaults to using the Reader if provided, otherwise tries to read from the
+// file
 func (t *TOMLLoader) Load(s interface{}) error {
-	filePath, err := getConfigPath(t.Path)
-	if err != nil {
-		return err
+	var r io.Reader
+	if t.Reader != nil {
+		r = t.Reader
+	} else {
```

Comment:
Can you add `else if t.Path != "" ` statement and return an error if both `t.Reader` and `t.Path` is not initialized?